src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1880,6 +1937,70 @@ void Partitioner::spatial(const std::string& func_name) {
     LOG_VERB("Done");
 }
 
+void Partitioner::dynamic(const std::string& func_name) {
+    ov::npuw::Function& f = P.functions.at(func_name);
+
+    // Support only attention at the time
+    if (f._tag != "attn") {
+        LOG_VERB("No dynamic handling be done to  " << func_name << " in model " << model->get_friendly_name()
+                                                    << "...");
+        return;
+    }
+
+    identifyDynamicParams(f);
+    if (!f._dynamic) {
+        LOG_WARN("Do dynamic ranges found in the ATTN block");
+        return;
+    }
+
+    // Apply transformation to the model. Note: only function body is modified
+    // Accumulate the reshape map
+    std::map<ov::Output<ov::Node>, ov::PartialShape> new_shapes;
+    for (auto&& p : f._dynamic->_inputs) {
+        ov::PartialShape dyn_shape = p.param->get_shape();  // Here it is yet static
+        dyn_shape[p.dim] = ov::Dimension();                 // ..and now is dynamic
+        new_shapes[p.param->output(0)] = std::move(dyn_shape);
+    }
+    // Mask
+    {
+        f._dynamic->_mask_shape = f._dynamic->_mask->get_shape();
+        ov::PartialShape dyn_shape = f._dynamic->_mask_shape;
+        dyn_shape[dyn_shape.size() - 1] = ov::Dimension();
+        new_shapes[f._dynamic->_mask->output(0)] = std::move(dyn_shape);
+    }
+    f._model->reshape(new_shapes);
+
+    // Patch Broadcast constants if there's any. If there's broadcast in the attention
+    // block, its shape argument is normally a precomputed Const (which would be
+    // an expression/a subgraph in the original dynamic IR). Since we retrofit
+    // dynamism into a static shape environment here, we need to patch it back.
+    for (auto &&op : f._model->get_ordered_ops()) {
+        if (!ov::is_type<ov::op::v3::Broadcast>(op)) {
+            continue;
+        }
+        // Inspect the constant
+        auto shape_source = op->input(1).get_source_output().get_node_shared_ptr();
+        if (!ov::is_type<ov::op::v0::Constant>(shape_source)) {
+            LOG_WARN("SDPA Broadcast's 2nd input is not Const: " << shape_source << ", skipping");
+            continue;
+        }
+
+        auto shape_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(shape_source);
+        auto shape_values = shape_const->cast_vector<int32_t>();
+        for (auto &&d : shape_values) {
+            if (d == f._dynamic->_mask_shape.back()) {
```

Comment:
this should be a context size, though it is not taken by the `function::Dynamic`